packfile.c

```diff
@@ -137,7 +137,7 @@ static int check_packed_git_idx(const char *path, struct packed_git *p)
 			munmap(idx_map, idx_size);
 			return error("wrong index v1 file size in %s", path);
 		}
-	} else if (version == 2) {
+	} else {
```

Comment:
Wait, what...?

What does this even *do*?

This change *must* be accompanied by *some* sort of explanation in the commit message. Without anything in the way of defending this change, this commit is no good.